File: geometry/golang/utils/approx-eq.go
package utils

import "math"

// An arbitrary maximum allowable difference in precision between floating-point numbers.
const Tolerance = 1e-07

// Tests for approximate equality between two 64-bit, floating-point numbers within an absolute
// or relative tolerance of Epsilon. An absolute tolerance is used for values
// less than or equal to 1.0. A relative tolerance is used for larger values.
func ApproxEq(x, y float64) bool {
	return math.Abs(x-y) <= Tolerance*math.Max(1.0, math.Max(math.Abs(x), math.Abs(y)))
}

File: geometry/python/utils/approx-eq.py
# EPSILON
# The maximum allowable difference in precision between floating-point numbers.

EPSILON = 0.000001

# approx_eq(number, number) -> boolean
# Tests for approximate equality between two floating-point numbers within an absolute
# or relative tolerance of EPSILON. An absolute tolerance is used for values
# less than or equal to 1.0. A relative tolerance is used for larger values.
# approx_eq(0.2, 0.19999999) -> True

def approx_eq(x, y):
    return abs(x - y) <= (EPSILON * max(1.0, abs(x), abs(y)))
